Plugins/Public/contraband/Main.cpp

```diff
@@ -0,0 +1,367 @@
+
+	foreach(lstHouseTag, INISECTIONVALUE, iter)
+	{
+		lstHouse.push_back(iter->scKey);
+		housesLoaded++;
+	}
+
+	INI_Reader ini;
+	if (ini.open(scPluginCfgFile.c_str(), false))
+	{
+		while (ini.read_header())
+		{
+			foreach(lstHouse, string, iter)
+			{
+				string scIter = *iter;
+				if (ini.is_header(scIter.c_str()))
+				{
+					while (ini.read_value())
+					{
+						if (ini.is_value("system"))
+						{
+							mapHouseAndSystems[CreateID(ini.get_value_string(0))] = scIter;
+							houseSystemsLoaded++;
+						}
+						else if (ini.is_value("contraband"))
+						{
+							mapHouseCargoList[scIter].push_back(CreateID(ini.get_value_string(0)));
+							contrabandLoaded++;
+						}
+					}
+				}
+				else if (ini.is_header("Laws"))
+				{
+					while (ini.read_value())
+					{
+						if (ini.is_value("house"))
+						{
+							mapHouseLawList[ini.get_value_string(0)] = stows(ini.get_value_string(1));
+							lawsLoaded++;
+						}
+					}
+				}
+			}
+		}
+		ini.close();
+
+		ConPrint(L"Contraband: Houses Loaded: %u \n", housesLoaded);
+		ConPrint(L"Contraband: House Systems Loaded: %u \n", houseSystemsLoaded);
+		ConPrint(L"Contraband: Laws Loaded: %u \n", lawsLoaded);
+		ConPrint(L"Contraband: Illegal Items Loaded: %u \n", contrabandLoaded);
+	}
+
+	string goods = R"(..\data\equipment\select_equip.ini)";
+	if (ini.open(goods.c_str(), false))
+	{
+		while (ini.read_header())
+		{
+			if (ini.is_header("Commodity"))
+			{
+				uint ids_name;
+				uint nick_id;
+				while (ini.read_value())
+				{
+					if (ini.is_value("nickname"))
+					{
+						nick_id = CreateID(ini.get_value_string());
+					}
+					else if (ini.is_value("ids_name"))
+					{
+						ids_name = ini.get_value_int(0);
+					}
+				}
+				mapItems[nick_id] = ids_name;
+			}
+		}
+		ini.close();
+		ConPrint(L"Contraband: Loaded Items: %u\n", mapItems.size());
+	}
+	HkLoadStringDLLs();
+	
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void ContrabandWarning(uint iClientID, wstring wscText, ...) // WE WANT BOLD RED
+{
+	wchar_t wszBuf[1024 * 8] = L"";
+	va_list marker;
+	va_start(marker, wscText);
+	_vsnwprintf(wszBuf, sizeof(wszBuf) - 1, wscText.c_str(), marker);
+
+	wstring wscXML = L"<TRA data=\"65281\" mask=\"-1\"/><TEXT>" + XMLText(wszBuf) + L"</TEXT>";
+	HkFMsg(iClientID, wscXML);
+}
+
+static void CheckCargo(int iClientID)
+{
+	if (set_lawsContrabandEnabled)
+	{
+		uint iSystemID;
+		pub::Player::GetSystem(iClientID, iSystemID);
+
+		int iHoldSize;
+		list<CARGO_INFO> lstCargo;
+		HkEnumCargo((const wchar_t*)Players.GetActiveCharacterName(iClientID), lstCargo, iHoldSize);
+
+		if (mapHouseAndSystems[iSystemID] != mapHouseAndSystems[iClient_systems[iClientID]] || mapHouseAndSystems.find(iClient_systems[iClientID]) == mapHouseAndSystems.end())
```

Comment:
In my testing, everything seems to work as intended. It will only ever alert you when you enter a new sector / undock in a sector after logging in. I ran in debug mode numerous times and I failed to find any errors or exceptions when leaving/entering sectors.